week3/predict_citibike.Rmd

```diff
@@ -0,0 +1,391 @@
+---
+title: "predict_citibike"
+author: "Sushobhan Parajuli"
+date: "`r Sys.Date()`"
+output: html_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+```{r}
+library(tidyverse)
+library(dplyr)
+library(scales)
+library(modelr)
+library(lubridate)
+```
+
+```{r}
+# import the holidays date
+# join holidays with the trips_per_day
+
+holidays <- read_csv("US Bank holidays", col_names = c("row", "ymd", "holiday"))
+
+# import the data "trips_per_day.tsv"
+# mutate a column "is_weekend" and "is_holiday"
+trips_per_day <- read_tsv("trips_per_day.tsv") %>% 
+  left_join(holidays, by = "ymd") %>%
+  mutate(is_holiday = as.numeric(!is.na(holiday))) %>%
+  mutate(is_weekend = as.numeric(wday(ymd)==c(1,7))) %>%
+  select(-c(row, holiday))
+
+trips_per_day
+```
+
+```{r}
+# split the data into train and test sets
+
+set.seed(10)
+
+num_days <- nrow(trips_per_day)
+frac_train <- 0.9
+num_train <- floor(num_days * frac_train)
+
+# randomly sample rows for the training set 
+ndx <- sample(1:num_days, num_train, replace=F)
+
+# used to fit the model and validate
+trips_per_day_rest <- trips_per_day[ndx, ]
+
+# used to test the fit
+trips_per_day_test <- trips_per_day[-ndx, ]
+
+trips_per_day_rest
+trips_per_day_test
+```
+
+```{r}
+# plot the relationship of variables with num_trips
+
+trips_per_day_rest %>% ggplot() + geom_point(aes(x = is_weekend, y = num_trips))
+trips_per_day_rest %>% ggplot() + geom_point(aes(x = is_holiday, y = num_trips))
+trips_per_day_rest %>% ggplot() + geom_point(aes(x = snow, y = num_trips))
+trips_per_day_rest %>% ggplot() + geom_point(aes(x = tmax, y = num_trips))
+trips_per_day_rest %>% ggplot() + geom_point(aes(x = tmin, y = num_trips))
+trips_per_day_rest %>% ggplot() + geom_point(aes(x = prcp, y = num_trips))
+```

```

Comment:
What do these plots tell us? Is it possible the way we're plotting these variables is affecting how we think about them?